lib/api_callable.js

```diff
@@ -168,6 +168,12 @@ function addTimeoutArg(aFunc, timeout, otherArgs, abTests) {
     options.deadline = new Date(now.getTime() + timeout);
     var metadata =
       otherArgs.metadataBuilder ? otherArgs.metadataBuilder(abTests) : null;
+    var headers = otherArgs.headers || {};
+    for (var key in headers) {
+      if (headers.hasOwnProperty(key)) {
+        metadata.set(key, headers[key]);
```

Comment:
That depends on the logic of analyzing the HTTP headers in our server. If `otherArgs.headers` specify `x-goog-api-client` value, I think `Metadata#add` will include both values separately, like
```
x-goog-api-client: gl-node/...
x-goog-api-client: (anything specified from users)
```

I'm afraid this could confuse our analysis server. For safety, I think it's better to avoid `Metadata#add`.

Instead of my suggestion above, another idea is to simply exclude well-known headers from the user-supplied data; and the only known header for us is just `x-goog-api-client`, so:
```javascript
for (var key in headers) {
  if (key !== 'x-goog-api-client' && headers.hasOwnProperty(key)) {
    metadata.set(key, headers[key]);
  }
}
```

this would work as expected.